theano/gof/compiledir.py

```diff
@@ -146,7 +147,7 @@ def short_platform(r=None, p=None):
 compiledir_format_dict['short_platform'] = short_platform()
 compiledir_format_keys = ", ".join(sorted(compiledir_format_dict.keys()))
 default_compiledir_format = ("compiledir_%(short_platform)s-%(processor)s-"
-                             "%(python_version)s-%(python_bitwidth)s")
+                             "%(python_version)s-%(python_bitwidth)s-%(device)s")
```

Comment:
That isn't good. I understand that this fix your problem, but this isn't the right fix. I'm ok to allow the user to use the device in the compiledir, but we should not change the default version. We should fix the underling problem.

Most of the time, this is a bad detection by g++ of the processor. We already have a few work around g++ bug in Theano. What if you processor?

Could try with the normal compiledir and with this flag:

gxx.cxxflags = -march=corei7

What is your g++ vesrion? Your exact CPU error?

Another user reported similar problem recently:

https://groups.google.com/forum/#!topic/theano-users/w-OIYeLSlZk